system/View/View.php

```diff
@@ -142,7 +142,7 @@ public function __construct(ViewConfig $config, string $viewPath = null, FileLoc
 		$this->loader   = $loader ?? Services::locator();
 		$this->logger   = $logger ?? Services::logger();
 		$this->debug    = $debug ?? CI_DEBUG;
-		$this->saveData = $config->saveData ?? null;
+		$this->saveData = (bool) $config->saveData;
```

Comment:
Correct me if I'm wrong. But in `View` config there is really no `$saveData` property so it is now added and yet in here it is used all the while? I think we still need the fallback to null here.

`$this->saveData = (bool) ($config->saveData ?? null);`

Reply:
We added it in **system/Config/View.php** so it will for sure always exist. Casting it to `bool` is there as a safeguard in case someone set it to `null` or `''` or `'monkeySoup'` or something in their **app/Config/View.php**.

Reply:
Hmm so this was a mishap from before. I think the `??` should still remain because if a user doesn't update his view config, then `$saveData` will still not be there and PHP will complain of an undefined property. Since we're casting to bool anyway, why not change the fallback to false, then cast to bool for safety.
```suggestion
		$this->saveData = (bool) ($config->saveData ?? false);
```

Reply:
I think we're talking past each other. `$saveData` has always been in **app/Config/View.php**, typed as `boolean`. That file extends **system/Config/View.php** where `$saveData` was missing (this PR adds it). Either way, it will always be present and always be a boolean so the null coalescence is unnecessary.

Reply:
Ah. See it now. Thanks for the clarification.